lighthouse-core/lib/page-functions.js

```diff
@@ -303,6 +303,39 @@ function getNodeLabel(node) {
   return tagName;
 }
 
+/**
+ * RequestIdleCallback shim that calculates the remaining deadline time in order to avoid a potential lighthouse
+ * penalty for tests run with simulated throttling. Reduces the deadline time to (50 - safetyAllowance) / cpuSlowdownMultiplier to
+ * ensure a long task is very unlikely if using the API correctly.
+ * @param {number} cpuSlowdownMultiplier
+ * @return {null}
+ */
+/* istanbul ignore next */
+function wrapRequestIdleCallback(cpuSlowdownMultiplier) {
+  const safetyAllowanceMs = 10;
+  const maxExecutionTimeMs = Math.floor((50 - safetyAllowanceMs) / cpuSlowdownMultiplier);
+  const nativeRequestIdleCallback = window.requestIdleCallback;
+  window.requestIdleCallback = (cb) => {
+    const cbWrap = (deadline, timeout) => {
+      const start = Date.now();
```

Comment:
will `performance.now` work too?

Reply:
it will, we're like split 50/50 throughout the smokes on Date.now vs. performance.now

Reply:
I think `Date.now()` is faster than `performance.now()` thats why i chose it, but i dont know if it matters all that much in this case. Do you want me to refactor to `performance.now()`?

Reply:
I don't think it's necessary @LukasAuerMstage but @connorjclark were you hoping to slowly standardize on performance.now?